### PR TITLE
feat: handle didcomm qr codes

### PR DESCRIPTION
**Pull request by maycon-mello**

### CODE REVIEW

**Comment by maycon-mello** [comment]
Would be good to add some tests on it, maybe we can push as separate PR

**Comment by maycon-mello** [comment]
Relates to https://github.com/docknetwork/react-native-sdk/pull/113

**Comment by maycon-mello** [comment]
image
image

**Comment by maycon-mello** [comment]
@Jayteekay will add some unit tests to it

**Comment by SamHellawell** [comment]
so does this go like QR -> notifications -> view instead of QR -> accept/decline -> view? guessing we dont have that state ready just yet, but that would be the ideal flow

also, are you submitting the ACK response to the relay after scanning and import was successful? we dont want the QRs to be scannable twice

**Comment by maycon-mello** [comment]
> 



> so does this go like QR -> notifications -> view instead of QR -> accept/decline -> view? guessing we dont have that state ready just yet, but that would be the ideal flow
> 
> also, are you submitting the ACK response to the relay after scanning and import was successful? we dont want the QRs to be scannable twice

`accept/decline`  is not implemented. let me create a jira ticket for that.
`submitting the ACK response to the relay after scanning`, this part is missing too. let me create another follow up jira for this one too

**Comment by SamHellawell** [comment]
>this part is missing too. let me create another follow up jira for this one too

itll take a few minutes to add a call to ACK, probably best just to do here. we dont want unencrypted messages containing VCs lingering longer than they have to

**Comment by maycon-mello** [comment]
> > this part is missing too. let me create another follow up jira for this one too
> 
> itll take a few minutes to add a call to ACK, probably best just to do here. we dont want unencrypted messages containing VCs lingering longer than they have to

sure, will need to add a new method on the relay service rpc service. we can do on this PR

File: src/features/qr-code-scanner/qr-code.js [hunk 1]
```diff
@@ -399,7 +402,38 @@ export async function relayServiceHandler(json) {
...
+  await relayServiceMessageHandler({
+    message,
+    callback: navigateBack,
+  });
+
+  const messageId = data.match(
+    /(?<=https:\/\/relay\.dock\.io\/read\/)[a-f0-9]+/,
+  )?.[0];
+
+  if (messageId) {
```
Comment (SamHellawell) [inline src/features/qr-code-scanner/qr-code.js:425]: at this point has the message been fully processed and converted into an imported credential?

**Review by SamHellawell — APPROVED** [review]
lgtm if the quick question i posted is a "yes" answer - which i think it is

**Pull request merged**